### ISSUE TITLE
Recent change to GTK File Open dialoge rendered it unusable for UFO 'files'

### ISSUE THREAD

**Issue by alerque** [linebender/runebender#108]
Some recent set of changes added a filter option to the GTK File Open dialogues, and the app is now looking to open things that it understands. I assume it is looking for UFO 'files'. The problem is UFOs are not files, they are a directory structure! As such I can find them in the chooser, but when I hit "Open" it just opens the directory in the chooser and never actually opens them in the app. I can't open an UFO at all through the dialogue, the only way I can open a UFO now is to launch the app passing the UFO name as the first argument.

**Comment by cmyr** [comment]
good to know, thanks. Quarantine has me away from a machine running GTK, so I've been less diligent in testing recently. Will report this back up to druid.

Also if you're curious to try a fix, you could edit `menu.rs`, L68, and replace:

```rust
FileDialogOptions::new().allowed_types(vec![UFO_FILE_TYPE])
```

with

```rust
FileDialogOptions::new().allowed_types(vec![UFO_FILE_TYPE]).select_directories(true)
``

and see if that changes anything?

**Comment by alerque** [comment]
That has the opposite affect. `.select_directories(true)` complains because it says it is expecting 0 args, and changing it to `.select_directories()` renders directories completely un-selectable (and hence un-navigable).

**Comment by cmyr** [comment]
hmmmm. okay, thanks!

**Comment by Finnerale** [comment]
Copying my comment from [druid#947](https://github.com/xi-editor/druid/issues/947):

Selecting directories works perfectly fine when not using `allowed_types`.

Using `allowed_types` + `select_directories` is 'broken' indeed, but I do not think we can do anything about that on our end.

The problem with the combination is that you can only navigate bookmarks, but not 'go into' any directories in the dialog because they are filtered out. But if you have a bookmark that gets you to a folder containing a `my_font.ufo` directory than that works perfectly fine (it is selectable).

Maybe we should just disable filtering directories for now?

**Comment by madig** [comment]
The traditional way on non-macOS is to set the filter to "metainfo.plist", so you have to go into the folder and select that file.

**Comment by xStrom** [comment]
If the goal is to select directories then `select_directories()` must definitely be used, otherwise this won't work on Windows. Also Windows only has file filtering via the OS provided functionality.

As I wrote in [druid#947](https://github.com/xi-editor/druid/issues/947) I think the easiest solution is to have the druid GTK implementation ignore `allowed_types` when using `select_directories`. Then it will work like Windows and you can choose directories, even if not filtered.

**Comment by jansol** [comment]
At the moment the open dialog seems to simply crash the whole application:
```
(runebender:18818): GLib-GIO-ERROR **: 02:32:12.988: No GSettings schemas are installed on the system
Trace/breakpoint trap (core dumped)
```

Might also be caused by something missing from my nix-shell environment though.

**Comment by davelab6** [comment]
I wonder if runebender can understand what to do when asked to open a `metainfo.plist` (and maybe for convenience also a `fontinfo.plist` or `.glif` file within a UFO)?

**Comment by alerque** [comment]
As mentioned [here](https://github.com/linebender/runebender/issues/224#issuecomment-782827276) apparently Windows suffers from the same issue an Linux. I guess only macOS is happy pretending directories are "files".

**Comment by xorgy** [comment]
> As mentioned [here](https://github.com/linebender/runebender/issues/224#issuecomment-782827276) apparently Windows suffers from the same issue an Linux. I guess only macOS is happy pretending directories are "files".

Yeah, macOS is the only one of these platforms where file some formats have the extension but can be either a file or a directory (see `.pkg`). It is pretty gnarly. It'd be like recognizing `.ufo` as a directory format, but as a file format if it's a gzipped tarball as well.
